pkg/deploy/controller/deployment/factory.go

```diff
@@ -122,6 +126,15 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 	}
 }
 
+// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
+func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
+	switch strategy.Type {
+	case deployapi.DeploymentStrategyTypeCustom:
+		util.MergeInto(pod.Labels, strategy.Labels, 0)
```

Comment:
I think if we are going to set labels and annotations on spec.strategy,
they should apply to pods created by the strategy (hooks + process) equally.

On Mon, Nov 2, 2015 at 2:25 PM, Dan Mace <notifications@github.com> wrote:

> In pkg/deploy/controller/deployment/factory.go
> <https://github.com/openshift/origin/pull/5149#discussion_r43669338>:
>
> > @@ -122,6 +126,15 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
> >  	}
> >  }
> >
> > +// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
> > +func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
> > +	switch strategy.Type {
> > +	case deployapi.DeploymentStrategyTypeCustom:
> > +		util.MergeInto(pod.Labels, strategy.Labels, 0)
>
> I'd like at least one other person (@smarterclayton
> <https://github.com/smarterclayton> @liggitt <https://github.com/liggitt>
> @kargakis <https://github.com/kargakis>) to validate my premise (that we
> can apply labels/annotations to all deployer pods). If all agree, you can
> remove the setPodAttributes abstraction.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5149/files#r43669338>.
>